### PR TITLE
Apply an init script from a gradle.properties script.

### PR DESCRIPTION
**Pull request by ethankhall**
Adds a new property that will be parsed from the gradle.properties named `org.gradle.init.url` (for now).

This PR should be rebased against #1022 to use the http cache.

This will allow users to specify a URL that will be applied before any init scripts.

Notifying @alkemist  @oehme @adammurdoch, who probably all have a lot of feedback about this.

### CODE REVIEW

File: subprojects/core/src/main/java/org/gradle/StartParameter.java [hunk 1]
```diff
@@ -90,6 +81,7 @@
     private boolean continuous;
     private List<File> includedBuilds = new ArrayList<File>();
     private boolean buildScan;
+    private String bootstrapInitScriptUrl;
```
Comment (oehme) [inline subprojects/core/src/main/java/org/gradle/StartParameter.java]: I wouldn't add another property but  reuse the existing list. Just like you can define properties both on the cli and in a properties file, you'd be able to define init script(s) in both places.

Comment (ethankhall) [inline subprojects/core/src/main/java/org/gradle/StartParameter.java]: I had originally tried to do this, but with a URL we don't have access to the dependency resolution stuff yet when the StartParameter is created. This object also seems to be more of a model without much of any logic in it and I didn't want to change that. We could change the list of files of init scripts to be a String or URL to we don't need any resolution logic but that's a change to the public (incubating) API. Thoughts?

Comment (oehme) [inline subprojects/core/src/main/java/org/gradle/StartParameter.java]: I see what you mean now. We could potentially change the list to contain `Object`s which are evaluated later using the resolver. That way you could also use `--init-script someUrl` when you want to quickly test something. I'll let the others chime in on this too.

Comment (ethankhall) [inline subprojects/core/src/main/java/org/gradle/StartParameter.java]: We could also have two lists, one for provided init scripts(ones located in init.d folders) and user init scripts. But I like your idea better. I can take that path and fix this up using that until others chime in.

**Comment by oehme** [comment]
As discussed via Slack as well as with @adammurdoch, we don't want to add more hooks to define init scripts. They are decorative in nature and so should not be required to run a build. If something is part of the build defintion, then it is better applied in `settings.gradle` using a remote script. See #745

**Pull request closed without merging**